### PR TITLE
fix: cli azure select sub

### PR DESCRIPTION
**Pull request by xiaolang124**
1. Add CLI azure select sub logic.
image

### CODE REVIEW

File: packages/cli/src/commonlib/azureLogin.ts [hunk 1]
```diff
@@ -465,24 +459,57 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
     }
   }
 
-  // TODO add login and select subscription logic later
-  getSelectedSubscription(triggerUI=false): Promise<SubscriptionInfo | undefined> {
-    if (AzureAccountManager.codeFlowInstance.account) {
+  async getSelectedSubscription(triggerUI = false): Promise<SubscriptionInfo | undefined> {
+    if (triggerUI) {
+      if (!AzureAccountManager.codeFlowInstance.account) {
+        await this.login(false);
```
Comment (swatDong) [inline packages/cli/src/commonlib/azureLogin.ts:465]: Why different from vscode, `true` or `false`?

File: packages/cli/src/commonlib/azureLogin.ts [hunk 2]
```diff
@@ -465,24 +459,57 @@ export class AzureAccountManager extends login implements AzureAccountProvider {
...
+      }
+      if (AzureAccountManager.codeFlowInstance.account && !AzureAccountManager.subscriptionId) {
+        const subscriptionList = await this.listSubscriptions();
+        if (!subscriptionList || subscriptionList.length == 0) {
```
Comment (swatDong) [inline packages/cli/src/commonlib/azureLogin.ts:469]: `===`
Please fix all in this file...

File: packages/cli/src/commonlib/azureLogin.ts (hunk 1 above)
Comment (xiaolang124) [reply packages/cli/src/commonlib/azureLogin.ts:465]: Now CLI will not ask user to confirm login or show learn more button.

File: packages/cli/src/commonlib/azureLogin.ts (hunk 2 above)
Comment (xiaolang124) [reply packages/cli/src/commonlib/azureLogin.ts]: Fixed.

**Review by swatDong — APPROVED** [review]

**Pull request merged**